ApplicationLibCode/ProjectDataModel/Summary/RimSummaryPlot.cpp

```diff
@@ -1977,22 +1978,118 @@ void RimSummaryPlot::handleDroppedObjects( const std::vector<caf::PdmObjectHandl
                 auto summaryCase = RiaSummaryTools::summaryCaseById( summaryAdr->caseId() );
                 if ( summaryCase )
                 {
-                    auto* newCurve = new RimSummaryCurve();
+                    addNewCurveY( summaryAdr->address(), summaryCase );
+                }
+            }
+            continue;
+        }
+
+        auto addressCollection = dynamic_cast<RimSummaryAddressCollection*>( obj );
```

Comment:
The RimSummaryPlot file is large. Consider moving parts of the following code into a separate tools file